### ISSUE TITLE
Regression: line_length does not trigger error when ignores_function_declarations is enabled

### ISSUE THREAD

**Issue by lukjj-public-pega** [realm/SwiftLint#6347]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Configuration:

```yml
line_length:
  warning: 100
  error: 100 
  ignores_function_declarations: true
```

Source file:
```swift
func test() {
    try FileManager.default.url(for: .cachesDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
}
```

Xcode version:
```bash
% xcodebuild -version
Xcode 26.1.1
Build version 17B100
```

I tested portable versions from GitHub. When using `0.62.2`:
```bash
./swiftlint --config ./.swiftlint.yaml
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

Latest working version was `0.59.1`:
```bash
./swiftlint --config ./.swiftlint.yaml
<path removed>/test.swift:2:1: error: Line Length Violation: Line should be 100 characters or less; currently it has 110 characters (line_length)
Done linting! Found 1 violation, 1 serious in 1 file.
```

Setting `ignores_function_declarations` to false resolves the issue on `0.62.2`.
